components/eam/src/physics/cam/modal_aer_opt.F90

```diff
@@ -1422,63 +1206,79 @@ subroutine calc_refin_complex (ncol, ilw,                   & ! in
 
     integer :: icol
 
-   crefin(:ncol) = (0._r8, 0._r8)
-   dryvol(:ncol) = 0._r8
+    if ((lwsw /= 'lw') .and. (lwsw /= 'sw')) call endrun()
```

Comment:
Thanks for adding this check. Can you please add an error messages in the endrun call to be explicit about what went wrong? For example:
```
character(len=100) :: err_msg
if ((lwsw /= 'lw') .and. (lwsw /= 'sw')) then
    write(err_msg,*)'calc_refin_complex is called with ', lwsw, ', it should be called with either lw or sw'
    call endrun(err_msg)
endif
```